File: bubble.cpp
#include <bits/stdc++.h>
using namespace std;

void display(vector<int> A, int n)
{
    cout << endl;
    for (auto i : A)
        cout << i << " ";
}

void bubbleSort (int arr[], int n){
    int i ,j ;
    bool swapped;
    for(i=0;i<n-1;i++){
        swapped = false;

    for (j=0;j<n-i-1;j++){
        if(arr[j] > arr[j+1]){
            swap(arr[j],arr[j+1]);
            swapped =true;
        }
    }
    if (swapped == false)
    break;
    }
}
void printArray(int arr[],int size){
    int i;
    for(i=0;i < size;i++)
    cout << " "<< arr[i];
}

int main()
{
    int arr[] = {5,3,1,9,8,2,4,7};
    int N =sizeof(arr)/sizeof(arr[0]);
    bubbleSort(arr,N);
    cout<< "Sorted array";
    printArray(arr,N);
    return 0; 
}

File: Bubble.java
// Optimized java implementation of Bubble sort

import java.io.*;

class GFG {
	
	// An optimized version of Bubble Sort
	static void bubbleSort(int arr[], int n)
	{
		int i, j, temp;
		boolean swapped;
		for (i = 0; i < n - 1; i++) {
			swapped = false;
			for (j = 0; j < n - i - 1; j++) {
				if (arr[j] > arr[j + 1]) {
					
					// Swap arr[j] and arr[j+1]
					temp = arr[j];
					arr[j] = arr[j + 1];
					arr[j + 1] = temp;
					swapped = true;
				}
			}

			// If no two elements were
			// swapped by inner loop, then break
			if (swapped == false)
				break;
		}
	}

	// Function to print an array
	static void printArray(int arr[], int size)
	{
		int i;
		for (i = 0; i < size; i++)
			System.out.print(arr[i] + " ");
		System.out.println();
	}

	// Driver program
	public static void main(String args[])
	{
		int arr[] = { 64, 34, 25, 12, 22, 11, 90 };
		int n = arr.length;
		bubbleSort(arr, n);
		System.out.println("Sorted array: ");
		printArray(arr, n);
	}
}

// This code is contributed
// by Nikita Tiwari.
